src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,9 +371,37 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (e.NewValue is TextWrapping textWrapping && (textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
+			{
+				if (_textBoxView != null)
+				{
+					//store  the value configured on the wrapper _textBoxView to restore if user changes from Wrap to UnWrap
+					_isSingleLine = _textBoxView.IsSingleLine;
+
+					//just change if IsSingleLine equals true, to avoid invoke the propertychange in the wrong way
+					if (_textBoxView.IsSingleLine)
+					{
+						//this must be setted to false to Wrap the text
+						_textBoxView?.SetSingleLine(false);
+					}
+				}
+			}
+			else
+			{
+				//just check if the "user value" was stored 
+				if (_isSingleLine.HasValue)
+				{
+					//return the value configured by user
+					_textBoxView.SetSingleLine(_isSingleLine.Value);
+
+					//reset the value
+					_isSingleLine = null;
+				}
+			}
```

Comment:
@aesteves900 you do not need to store the value here, as `TextBoxView` is only manipulated by `TextBox` (unless you've seen otherwise). If the text wrapping does not allow multiline, invoke `SetSingleLine(true)` otherwise `SetSingleLine(false)`.

```suggestion
		   if (_textBoxView != null && e.NewValue is TextWrapping textWrapping)
		   {
		        var needsWrapping = textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords;
               _textBoxView.SetSingleLine(!needsWrapping);
		   }
```